File: 2026 05월/05월 3주차/JeongEon8/03/[PGS - Lv2] 귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    map<int, int> m;
    for(int i = 0; i < tangerine.size(); i++) {
        m[tangerine[i]]++;
    }
    
    vector<int> v;
    for(auto u : m) {
        v.push_back(u.second);
    }
    
    sort(v.begin(), v.end(), greater<>());
    
    int cnt = 0;
    for(int i = 0; i < v.size(); i++){
        if(cnt >= k) {
            break;
        }
        
        answer++;
        cnt += v[i];
    }
    
    return answer;
}

File: 9월 2주차/hyuksanglee/01/[PGS - Lv2] 귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;

        Map<Integer, Integer> map = new HashMap<>();

        for (int num : tangerine) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        List<Integer> list = new ArrayList<>(map.values());

        list.sort(Comparator.reverseOrder());

        int total = 0;

        for (int n : list) {
            total += n;
            answer++;

            if (total >= k) {
                break;
            }
        }

        return answer;
    }
}
